server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1760,8 +1761,22 @@ public void updateSegmentMetadata(final Set<DataSegment> segments)
           @Override
           public Void inTransaction(Handle handle, TransactionStatus transactionStatus) throws Exception
           {
+            final String updatePayload = StringUtils.format("UPDATE %s SET payload = :payload WHERE id = :id", dbTables.getSegmentsTable());
+            final PreparedBatch batch = handle.prepareBatch(updatePayload);
+
             for (final DataSegment segment : segments) {
-              updatePayload(handle, segment);
+              String id = segment.getId().toString();
+              byte[] payload = jsonMapper.writeValueAsBytes(segment);
```

Comment:
I think keeping the `inTransaction` code block concise and focused to only transaction-related things will generally reduce the DB transaction duration and make the code a bit more readable. For example, this serialization can be moved above  `connector.getDBI().inTransaction()` and collected in a list and here we can just iterate over the list and do a simple `bind()`

Reply:
+1

Reply:
Great idea, will do.